src/main/java/org/terasology/additionalRails/components/ActivableCartComponent.java

```diff
@@ -0,0 +1,15 @@
+package org.terasology.additionalRails.components;
+
+import org.terasology.entitySystem.Component;
+
+/**
+ * Component which makes the cart able to receive {@link org.terasology.additionalRails.events.CartActivatedEvent}.
+ * Every cart, which want to use the Activator Rail, must contain this component.
+ * @author Aleksander Wójtowicz <anuar2k@outlook.com>
+ */
+public class ActivableCartComponent implements Component {
```

Comment:
I'm debating if this is really necessary. You could just send an event to all the carts that pass over the segment. Not sure why you need a special component that denotes if a cart can be activated.

Reply:
I've wanted to make the cart know if it is still on/above activator rail (some UpdateSubscriberSystem would simply check, if eventCalled [or rather aboveActivator, i'll rename it] is true, to do some continous action only if the cart is on the rail).

There are three ways of doing it:
- We can send the CartActivatedEvent only once, make the cart have the ActivableCartComponent with that boolean (so we leave it as it is now), or:
- send CartActivatedEvent when cart enters the rail and CartDeactivatedEvent, when it leaves the activator, or:
- we don't want to make the cart know when it leaves the rail and just use CartActivatedEvent (therefore the cart will have no way of checking if it is still activated or not).

I'm leaving this decision up to you. And thanks for the review! :)

Reply:
I have updated my work and went by the second way described above. After the rework, Activator Rail's behavior was simplified a lot.